src/v/datalake/tests/record_multiplexer_test.cc

```diff
@@ -158,7 +158,13 @@ class RecordMultiplexerTestBase
             total_split_batches += subset.size();
             auto reader = model::make_memory_record_batch_reader(
               std::move(subset));
-            mux.multiplex(std::move(reader), model::no_timeout, as).get();
+            mux
```

Comment:
Worth adding a simple unit test? I guess the interruptions test caught this, but it seems like a surprising way to catch this bug